docs/build_from_source.md

```diff
@@ -0,0 +1,98 @@
+# Building from source
+
+This document gives information how to build docker images from source with different variants.
+
+## Prerequisites
+
+1. [Docker Engine](https://docs.docker.com/engine/)
+1. Ubuntu 20.04 or RedHat 8.7 host
+1. make
+1. bash
+
+## Makefile and building
+
+Makefile located in root directory of this repository contains all targets needed to build docker images and binary packages.
+
+It contains `docker_build` target which by default builds multiple docker images:
+- `openvino/model_server:latest` - smallest release image containing only neccessary files to run model server on CPU, NCS and HDDL
+- `openvino/model_server:latest-gpu` - release image containing support for Intel GPU
+- `openvino/model_server:latest-nginx-mtls` - release image containing examplary NGINX MTLS configuration
+- `openvino/model_server-build:latest` - image with builder environment containing all the tools to build OVMS
+
+The `docker_build` target also prepares binary package to run OVMS as standalone application and shared library to link against user written C/C++ applications.
+
+```bash
+git clone https://github.com/openvinotoolkit/model_server
+cd model_server
+make docker_build
+tree dist/ubuntu
+````
+
+```
+dist/ubuntu
+├── Dockerfile.redhat
+├── Dockerfile.ubuntu
+├── drivers
+├── LICENSE
+├── Makefile
+├── ovms.tar.gz
+├── ovms.tar.gz.metadata.json
+├── ovms.tar.gz.sha256
+├── ovms.tar.xz
+├── ovms.tar.xz.metadata.json
+├── ovms.tar.xz.sha256
+└── thirdparty-licenses
+```
+
+## Building Options
+
+### `INSTALL_DRIVER_VERSION`
```

Comment:
old driver version can be installed without downloading first

Reply:
listed